lib/src/input_validation/validate_function.dart

```diff
@@ -1,19 +1,27 @@
 import 'package:advanced_calculation/src/input_validation/error_state.dart';
 import 'package:advanced_calculation/src/input_validation/pattern.dart';
+import 'package:advanced_calculation/src/parse/expression_parser.dart';
+import 'package:advanced_calculation/src/translator/translate_pattern.dart';
 
 import 'start_state.dart';
 import 'state.dart';
 
 class ValidateFunction {
+  ExpressionParser parser = ExpressionParser();
+
   static final List<String> validFunctions = ["ln","log","sin","cos","tan", "abs", "csc","sec", "cot", "sqrt", "sinh", "cosh", "tanh",
     "asin", "acos", "atan", "acsc", "asec", "acot", "csch", "sech", "coth", "ceil","asinh", "acosh", "atanh", "acsch", "asech",
-    "acoth", "floor", "round", "trunc", "fract"];
+    "acoth", "floor", "round", "trunc", "fract", "√"];
   static final List<String> multiParamFunctions = ["max", "min", "gcd", "lcm"];
-
+  static final List<String> operands = ['*','/','-','+','(',')','^',',','²'];
 
   List<String> _sanitizeInput(String input){
-    List<String> sanitizedInput = input.split(" ").where((item) => item.isNotEmpty).toList();
-    sanitizedInput.add('=');
+    String trimmed = input;
```

Comment:
String trimmed = input + " =" (NOTE: add space before = sign)

Reply:
this is just validating the exact string the user provides, so the = is not necessary